teachertool/src/components/PrintButton.tsx

```diff
@@ -1,17 +1,23 @@
 import * as React from "react";
-import { useRef } from "react";
 import { useReactToPrint } from "react-to-print";
 import { Toolbar } from "./Toolbar";
+import { stateAndDispatch } from "../state";
+import { showToast } from "../state/actions";
+import { makeToast } from "../utils";
+import { Strings } from "../constants";
 
 interface PrintButtonProps {
     printRef: React.RefObject<HTMLDivElement>;
-    onHandlePrint: () => void;
 }
 
-export const PrintButton: React.FC<PrintButtonProps> = ({ printRef, onHandlePrint }) => {
+export const PrintButton: React.FC<PrintButtonProps> = ({ printRef }) => {
+    const { state: teacherTool } = stateAndDispatch();
     const handlePrint = useReactToPrint({
         content: () => printRef.current,
-        onAfterPrint: onHandlePrint,
+        onPrintError: () => showToast(makeToast("error", lf("Unable to print evaluation results"), 2000)),
+        documentTitle: `MakeCode Project Evaluation Results - ${pxt.Util.sanitizeFileName(
+            teacherTool.projectMetadata?.name || Strings.UntitledProject
+        )}`,
```

Comment:
Can this be localized?

Reply:
I can definitely put this into `lf`, but I figure we'll likely not want all of it localized. I can definitely localize whatever we decide to put as the leading descriptor (the `MakeCode Project Evaluation Results` here), but I figured we wouldn't want to localize the name of the project.

Reply:
Typically with translations like this, you'd do lf("MakeCode Project Evaluation Results - {0}", pxt.Util.sanitizeFileName(...))

The stuff passed into the format won't get translated, but the initial string will be formatted with the additional context that something is going to get put in place of that {0}.